File: S1_AM_LoginRegister/app/src/main/java/com/example/s1_loginregister/MainActivity.java
package com.example.s1_loginregister;

import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.text.InputType;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText txtUsuario, txtClave;
    Button btnLogin;
    TextView lblRegistrar;
    ImageView imgEye;
    boolean isPasswordVisible = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txtUsuario = findViewById(R.id.txtusuario);
        txtClave = findViewById(R.id.txtclave);
        btnLogin = findViewById(R.id.button);
        lblRegistrar = findViewById(R.id.lblregistrar);
        imgEye = findViewById(R.id.imgEye);

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String usuario = txtUsuario.getText().toString();
                String clave = txtClave.getText().toString();

                if (usuario.isEmpty() || clave.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Campos vacíos", Toast.LENGTH_SHORT).show();
                } else {
                    validarUsuario(usuario, clave);
                }
            }
        });

        lblRegistrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, RegisterActivity.class);
                startActivity(intent);
            }
        });

        imgEye.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (isPasswordVisible) {

                    txtClave.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
                    imgEye.setImageResource(R.drawable.baseline_eye_off);
                } else {

                    txtClave.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD);
                    imgEye.setImageResource(R.drawable.baseline_eye_open);
                }

                isPasswordVisible = !isPasswordVisible;

                txtClave.setSelection(txtClave.getText().length());
            }
        });
    }

    private void validarUsuario(String usuario, String clave) {
        AdminSQLiteOpenHelper admin = new AdminSQLiteOpenHelper(this);
        SQLiteDatabase db = admin.getReadableDatabase();

        Cursor cursor = db.rawQuery(
                "SELECT * FROM usuarios WHERE usuario = ? AND clave = ?",
                new String[]{usuario, clave}
        );

        if (cursor.moveToFirst()) {
            Toast.makeText(this, "Inicio de sesión exitoso", Toast.LENGTH_SHORT).show();
            Intent intent = new Intent(MainActivity.this, BienvenidoActivity.class);
            startActivity(intent);
            finish();
        } else {
            Toast.makeText(this, "Usuario o contraseña incorrectos", Toast.LENGTH_SHORT).show();
        }

        cursor.close();
        db.close();
    }
}

File: S2_RegisterAlumnosUNS/app/src/main/java/com/example/s3_registeralumnosuns/MainActivity.kt
package com.example.s3_registeralumnosuns

import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    companion object {
        val listaAlumnos = mutableListOf<Alumno>()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editCodigo = findViewById<EditText>(R.id.editCodigo)
        val editApellidos = findViewById<EditText>(R.id.editApellidos)
        val editNombres = findViewById<EditText>(R.id.editNombres)
        val editCorreo = findViewById<EditText>(R.id.editCorreo)
        val btnRegistrar = findViewById<Button>(R.id.btnRegistrar)
        val btnVerRegistros = findViewById<Button>(R.id.btnVerRegistros)

        btnRegistrar.setOnClickListener {
            val alumno = Alumno(
                editCodigo.text.toString(),
                editApellidos.text.toString(),
                editNombres.text.toString(),
                editCorreo.text.toString()
            )
            listaAlumnos.add(alumno)
            editCodigo.text.clear()
            editApellidos.text.clear()
            editNombres.text.clear()
            editCorreo.text.clear()
        }

        btnVerRegistros.setOnClickListener {
            startActivity(Intent(this, ListaAlumnosActivity::class.java))
        }
    }
}
